Planit/View/Utils/PlanButtonStack.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  PlanButtonStack.swift
+//  Planit
+//
+//  Created by Shivangi on 06/12/22.
+//
+
+import SwiftUI
+
+struct PlanButtonStack: View {
+    
+    
+    @ObservedObject var viewModel: planViewModel
+    var body: some View {
+        ScrollView(.horizontal, showsIndicators: false){
+            HStack{
```

Comment:
Here, you could use a `ForEach` and iterate through all the button types.

Swift provides a protocol for enums where you can make your enum conform to `CaseIterable`, and then you would have `SelectedButton.allCases`  available so you could iterate through them. Then you can make changes to the enum, and your UI updates automatically!

`enum SelectedButton: String, CaseIterable {`